mars/errors.py

```diff
@@ -14,13 +14,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from .compat import six
+
 
 class MarsError(RuntimeError):
     """
     """
     def __init__(self, msg=None):
         self._err_data = msg
-        msg = repr(msg) if msg is not None else None
+        if msg is not None and not isinstance(msg, six.string_types):
+            msg = repr(msg)
```

Comment:
Is this used?

Reply:
This modification is aimed to yield better error message. The `repr('abc')` will become`"'abc'"`. For example:

```python
In [2]: raise Exception('abcd\n\tdef')
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-2-fc45df11d92c> in <module>
----> 1 raise Exception('abcd\n\tdef')

Exception: abcd
	def

In [3]: raise Exception(repr('abcd\n\tdef'))
---------------------------------------------------------------------------
Exception                                 Traceback (most recent call last)
<ipython-input-3-51832293106b> in <module>
----> 1 raise Exception(repr('abcd\n\tdef'))

Exception: 'abcd\n\tdef'
```

But if we revise the `ExecutionFailed` part according to the comment above, I think this optimization can be removed.

Reply:
Maybe it is better to put this logic into __str__ to reduce the cost of ``isinstance``.